src/FakeItEasy/MethodBaseExtensions.cs

```diff
@@ -18,5 +20,119 @@ public static string GetGenericArgumentsCSharp(this MethodBase method)
 
             return string.Concat("<", string.Join(", ", genericArguments.Select(type => type.FullNameCSharp()).ToArray()), ">");
         }
+
+        public static bool IsPropertyGetterOrSetter(this MethodBase method)
+        {
+            return method.IsPropertyGetter() || method.IsPropertySetter();
+        }
+
+        public static bool IsPropertySetter(this MethodBase method)
+        {
+            return method.IsSpecialName && method.Name.StartsWith("set_", StringComparison.Ordinal);
+        }
+
+        public static bool IsPropertyGetter(this MethodBase method)
+        {
+            return method.IsSpecialName && method.Name.StartsWith("get_", StringComparison.Ordinal);
+        }
+
+        public static string GetDescription(this MethodBase method)
+        {
+            var builder = new StringBuilder();
+
+            builder
+                .Append(method.DeclaringType)
+                .Append(".");
+
+            AppendMethodName(builder, method);
+
+            AppendParameterList(builder, method);
+
+            return builder.ToString();
+        }
+
+        private static void AppendMethodName(StringBuilder builder, MethodBase method)
+        {
+            if (IsPropertyGetterOrSetter(method))
+            {
+                builder.Append(method.Name.Substring(4));
+            }
+            else
+            {
+                builder.Append(method.Name);
+            }
+
+            builder.Append(method.GetGenericArgumentsCSharp());

```

Comment:
For consistency with how calls are formatted. Will need to be changed as part of #1047